### PR TITLE
Feat/sm layer detection

### PR DESCRIPTION
**Pull request by samuelOsborne**
This PR adds layer hit detection, along with a few refactors

Features:

- Layer hit check based on a layer name, x & y coordinate
- Added `getLayerBounds` API function. Pass a layer name and it returns a vec where the first 4 values correspond to x, y, width, height
- Added `perform_hit_check(&self, target: &str, x: f32, y: f32)` which returns `true` if the supplied target string (layer name) and x, y coordinate are within its bounding box, otherwise returns `false`

Refactors:

- Removed `postEventPayload` for web, replaced with individual functions
- Created and split events in to "Internal" events and "Events". This was to accommodate for the fact that we need different event definitions depending if we're managing the configuration file or at runtime. The was mostly the fact for pointer events where we declare a pointer event and a "target" property, but at runtime we send a pointer event with x and y coordinates.
- Split out the state machine demoes in to self-contained runnable examples from the bin folder.

### CODE REVIEW

**Comment by samuelOsborne** [comment]
Web tests

https://github.com/user-attachments/assets/5671c5ee-d45a-4550-900c-95027f330da3

**Comment by samuelOsborne** [comment]
iOS tests
https://github.com/user-attachments/assets/4ce9553e-3bbf-4f0e-909e-8387c097579e

**Comment by samuelOsborne** [comment]
Rust tests


https://github.com/user-attachments/assets/f2cdbc74-a596-4394-8a71-4dee36e5c2b2

File: dotlottie-rs/src/state_machine/events/mod.rs [hunk 1]
```diff
@@ -12,7 +12,21 @@ pub enum Event {
...
+    SetNumericContext { key: String, value: f32 },
+}
+
+#[derive(Debug, Clone)]
+pub enum InternalEvent {
+    Bool { value: bool },
+    String { value: String },
+    Numeric { value: f32 },
+    OnPointerDown { target: Option<String> },
+    OnPointerUp { target: Option<String> },
```
Comment (theashraf) [inline dotlottie-rs/src/state_machine/events/mod.rs:26]: [question]
The internal and external pointer event are very confusing, specially in the way they are defined. 
for example: OnPointerUp internally has a target of Option<String> but in the external takes x, y as f32. 
Maybe can you help me understand the reason for choosing this approach ?

File: dotlottie-rs/src/dotlottie_player.rs [hunk 2]
```diff
@@ -181,6 +181,15 @@ impl DotLottieRuntime {
         self.renderer.hit_check(layer_name, x, y).unwrap_or(false)
     }
 
+    pub fn get_layer_bounds(&self, layer_name: &str) -> Vec<f32> {
+        let result = self
+            .renderer
+            .get_layer_bounds(layer_name)
+            .unwrap_or((0.0, 0.0, 0.0, 0.0));
+
+        vec![result.0, result.1, result.2, result.3]
```
[the remark this answers is not in the document]
Comment (samuelOsborne) [reply dotlottie-rs/src/dotlottie_player.rs:190]: Is there a benefit to adding this? I feel like it makes the code harder to read

**Review by theashraf — APPROVED** [review]

**Pull request merged**